app/src/main/java/com/infomaniak/mail/ui/main/settings/ThreadModeSettingViewModel.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Infomaniak ikMail - Android
+ * Copyright (C) 2022-2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.settings
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.mail.data.cache.RealmDatabase
+import com.infomaniak.mail.data.cache.mailboxInfo.MailboxController
+import com.infomaniak.mail.di.IoDispatcher
+import com.infomaniak.mail.utils.AccountUtils
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ThreadModeSettingViewModel @Inject constructor(
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) : ViewModel() {
+
+    fun dropAllMailboxesContent() = viewModelScope.launch(ioDispatcher) {
+
+        RealmDatabase.closeMailboxContent()
+
+        MailboxController.getMailboxes().forEach { mailbox ->
+            RealmDatabase.deleteMailboxContent(mailbox.mailboxId, mailbox.userId)
+        }
+
+        AccountUtils.reloadApp?.invoke()
```

Comment:
Should the reload of the app be elsewhere than in a function that drops mailboxsContent? Like in the calling function right after the call to `dropAllMailboxesContent()`